static/app/components/teamSelector.tsx

```diff
@@ -156,6 +161,14 @@ function TeamSelector(props: Props) {
   const api = useApi();
   const {teams, fetching, onSearch} = useTeams();
 
+  const isMemberInvite =
+    organization.access?.includes('member:invite') &&
+    !organization.access?.includes('member:admin');
+  const membersCanOnlyInviteMemberTeams =
+    organization.allowMemberInvite && !organization.openMembership;
+  // If Member Invites are enabled but Open Membership is disabled, only allow members to invite to teams they are a member of
+  const memberTeamsOnly = isInviting && isMemberInvite && membersCanOnlyInviteMemberTeams;
```

Comment:
Since this logic seems generic and is derived entirely from the `organization` object, it could be a good candidate to extract into a utility function. There is probably a better name, but something like `organizationAllowsMemberInvitesOnlyWithinTeam`.